lib/cashbox/concern/persistable.rb

```
@@ -9,7 +9,7 @@ module Persistable
       extend SingleForwardable
 
       def_single_delegators :repository, :where, :all, :first, :find
-      def_instance_delegators :repository, :save, :destroy
+      def_instance_delegators :repository, :save, :save!, :destroy
```

why do you need a save and a save! ?

---

I take it to mean `save` is a safe "Will try, but if not, it'll recover." while `save!` is "There be dragons, if something goes wrong, you handle it."
Correct me if that's wrong, Jon

---

@julamula & @sftwninja the save & save! pattern is following a rails paradigm. In ActiveRecord, when you `save` it returns true/false if it worked. Then there's a `model.errors` object for you to do work with. This is great for returning errors from a controller. 

In a different setting (like a sidekiq job), we don't have the ability to send errors back since it's not in a request cycle. Instead we need to throw the errors so sidekiq can catch it, re-queue the job, and send the error to raygun.

---

oh neat!  💡